foundry/mouseInteractionManager.d.ts

```diff
@@ -1,213 +1,252 @@
    * @returns Can the action be performed?
    */
-  can(action: string, event: Event): boolean;
+  can(action: keyof this['permissions'], event: Event | PIXI.InteractionEvent): boolean;
 
   /**
-   * Activate a new set of listeners for click events on the target object
+   * Execute a callback function associated with a certain action in the workflow
+   * @param action - The action being attempted
+   * @param event  - The event being handled
    */
-  protected _activateClickEvents(): void;
+  callback(action: keyof this['callbacks'], event: Event | PIXI.InteractionEvent): unknown;
 
   /**
-   * Activate events required for handling a drag-and-drop workflow
+   * A reference to the possible interaction states which can be observed
    */
-  protected _activateDragEvents(): void;
+  get states(): typeof MouseInteractionManager['INTERACTION_STATES'];
 
   /**
    * Activate a set of listeners which handle hover events on the target object
    */
   protected _activateHoverEvents(): void;
 
+  /**
+   * Activate a new set of listeners for click events on the target object
+   */
+  protected _activateClickEvents(): void;
+
   /**
    * Deactivate event listeners for click events on the target object
    */
   protected _deactivateClickEvents(): void;
 
   /**
-   * Deactivate events required for handling drag-and-drop workflow.
+   * Activate events required for handling a drag-and-drop workflow
    */
-  protected _deactivateDragEvents(): void;
+  protected _activateDragEvents(): void;
+
   /**
-   * Handle mouse-down which trigger a single left-click workflow.
+   * Deactivate events required for handling drag-and-drop workflow.
    */
-  protected _handleClickLeft(event: Event): void;
+  protected _deactivateDragEvents(): void;
 
   /**
-   * Handle mouse-down which trigger a single left-click workflow.
+   * Handle mouse-over events which activate downstream listeners and do not stop propagation.
    */
-  protected _handleClickLeft2(event: Event): any;
+  protected _handleMouseOver(event: PIXI.InteractionEvent): unknown;
 
   /**
-   * Handle single right-click actions.
+   * Handle mouse-out events which terminate hover workflows and do not stop propagation.
    */
-  protected _handleClickRight(event: Event): void;
+  protected _handleMouseOut(event: PIXI.InteractionEvent): unknown;
 
   /**
-   * Handle double right-click actions.
+   * Handle mouse-down events which activate downstream listeners.
+   * Stop further propagation only if the event is allowed by either single or double-click.
    */
-  protected _handleClickRight2(event: Event): any;
+  protected _handleMouseDown(event: PIXI.InteractionEvent): unknown;
 
   /**
-   * Handle the cancellation of a drag workflow, resetting back to the original state
+   * Handle mouse-down which trigger a single left-click workflow.
    */
-  protected _handleDragCancel(event: Event): void;
+  protected _handleClickLeft(event: PIXI.InteractionEvent): unknown;
```

Comment:
I believe this always has a void return. Along with `_handleClickRight`, `_handleDragDrop`, and `_handleDragCancel`